src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -457,25 +457,39 @@ using namespace std;
         writer << "extern \"C\" void " << f->get_name() << "(void** inputs, void** outputs, "
                << "gpu::GPURuntimeContext* ctx);\n";
     }
-
     writer << "\n";
 
+    // This for loop creates a collection of functions that are called more than once
+    // and emitting them as globally callable functions.
+    // ops implement the is_functionally_identical method
     unordered_map<Node*, string> match_functions;
     for (shared_ptr<Function> current_function : pass_manager.get_state().get_functions())
     {
-        set<string> output_names;
-        for (shared_ptr<Node> op : current_function->get_results())
-        {
-            shared_ptr<descriptor::TensorView> tv = op->get_output_tensor_view();
-            output_names.insert(tv->get_tensor().get_name());
-        }
-        const list<shared_ptr<Node>>& tmp = current_function->get_ordered_ops();
+        list<shared_ptr<Node>> tmp = current_function->get_ordered_ops();
         if (tmp.size() < 2)
         {
             // Since we are comparing ops there must be at least two ops to proceed.
             continue;
         }
         vector<shared_ptr<Node>> op_list{tmp.begin(), tmp.end()};
+        unordered_map<const Node*, string> node_cache;
+        for (size_t i = 0; i < op_list.size(); i++)
```

Comment:
Here looks like n^2 complexity, I wonder if an unordered map with the string as key will help.

Reply:
fixed